spec/lib/hubspot/blog_spec.rb

```diff
@@ -1,116 +1,150 @@
 require 'timecop'
 
 describe Hubspot do
-  let(:example_blog_hash) do
-    VCR.use_cassette("blog_list", record: :none) do
-      url = Hubspot::Connection.send(:generate_url, Hubspot::Blog::BLOG_LIST_PATH)
-      resp = HTTParty.get(url, format: :json)
-      resp.parsed_response["objects"].first
-    end
-  end
-  let(:created_range_params) { { created__gt: false, created__range: (Time.now..Time.now + 2.years)  } }
-
   before do
     Hubspot.configure(hapikey: "demo")
-    Timecop.freeze(Time.local(2012, 'Oct', 10))
+    Timecop.freeze(Time.utc(2012, 'Oct', 10))
   end
 
   after do
     Timecop.return
   end
 
   describe Hubspot::Blog do
-
     describe ".list" do
-      cassette "blog_list"
-      let(:blog_list) { Hubspot::Blog.list }
+      it "returns a list of blogs" do
+        VCR.use_cassette("blog_list", record: :none) do
+          result = Hubspot::Blog.list
 
-      it "should have a list of blogs" do
-        blog_list.count.should be(1)
+          assert_requested :get, hubspot_api_url("/content/api/v2/blogs?hapikey=demo")
+          expect(result).to be_kind_of(Array)
+          expect(result.first).to be_a(Hubspot::Blog)
+        end
       end
     end
 
     describe ".find_by_id" do
-      cassette "blog_list"
+      it "retrieves a blog by id" do
+        VCR.use_cassette("blog_list", record: :none) do
+          id = 351076997
```

Comment:
Ideally, we wouldn't have this mystery id value and tests would be self-sufficient. By self-sufficient, I mean the test would include necessary setup (create a blog), request all blogs, and then teardown (delete the blog). This test is currently **not** self-sufficient because attempting to re-record this request would likely result in a 404 as the data hasn't been created on `api.hubapi.com`.

I've intentionally avoided tackling this right now as creating a blog isn't supported and I'm trying to keep this set of changes from growing.

Reply:
There are several places in the tests you’ll find things like this that rely on hard-coded values that reference some object created in the demo account on HubSpot. This is something I’ve been looking to address with the next version but I haven’t decided if it’s something that should just wait since all the RSpec tests need to be updated to version 3 anyway which is a significant update.